driver-sync/src/test/functional/com/mongodb/client/unified/UnifiedTestFailureValidator.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.client.unified;
+
+import com.mongodb.MongoClientSettings;
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.MongoClients;
+import com.mongodb.lang.Nullable;
+import org.bson.BsonArray;
+import org.bson.BsonDocument;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runners.Parameterized;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.util.Collection;
+
+public class UnifiedTestFailureValidator extends UnifiedTest {
+    private final String fileDescription;
+    private final String testDescription;
+    private Throwable setupException;
+
+    public UnifiedTestFailureValidator(final String fileDescription, final String testDescription, final String schemaVersion,
+                                       @Nullable final BsonArray runOnRequirements, final BsonArray entities, final BsonArray initialData,
+                                       final BsonDocument definition) {
+        super(schemaVersion, runOnRequirements, entities, initialData, definition);
+        this.fileDescription = fileDescription;
+        this.testDescription = testDescription;
```

Comment:
Good question.  First let me explain the reason for the parameters, in case that's not clear: on line 69 you'll see the annotation:

    @Parameterized.Parameters(name = "{0}: {1}")

That gives the test a nice name based on the file and test description, and the only way I know to do that is by passing them as parameters to the constructor.

The unused fields are just to avoid the unused parameter warnings, and in most cases we've added an annotation to the field:

    @SuppressWarnings("FieldCanBeLocal")

to suppress the warning for the unused field.  But I realize now there's a better way: just annotate the unused parameters directly with 

     @SuppressWarnings("unused")

which I've now done.  Let me know if you like that better.

Reply:
Thanks for the explanation! I think the new approach is cleaner 👍